Maple2.Database.Seed/Program.cs

```diff
@@ -52,18 +56,61 @@
 
 Console.WriteLine("Seeding complete!");
 
-
 void Seed(string type) {
     Stopwatch stopwatch = new();
 
-    Console.Write($"Seeding {type}... ");
+    Console.WriteLine($"Seeding {type}... ");
+
+    if (IsTableEmpty(type)) {
+        string fileLines = File.ReadAllText(Path.Combine(Paths.DB_SEEDS_DIR, $"{type}.sql"));
+        if (ExecuteSqlFile(fileLines)) {
+            Console.WriteLine($"{type} seeded in {stopwatch.ElapsedMilliseconds}ms");
+        } else {
+            Console.WriteLine($"Failed to seed {type}");
+        }
+        return;
+    }
 
-    string fileLines = File.ReadAllText(Path.Combine(Paths.DB_SEEDS_DIR, $"{type}.sql"));
-    if (ExecuteSqlFile(fileLines)) {
-        Console.Write($"finished in {stopwatch.ElapsedMilliseconds}ms");
-        Console.WriteLine();
+    Console.Write("Would you like to replace all data? (y/n): ");
+    string input = Console.ReadLine()?.Trim().ToLower();
+
+    if (input == "y") {
+        // Replace existing data
+        var foreignKeys = GetForeignKeyConstraints(type);
+        if (DropForeignKeyConstraints(foreignKeys)) {
+            TruncateTable(type);
+            string fileLines = File.ReadAllText(Path.Combine(Paths.DB_SEEDS_DIR, $"{type}.sql"));
+            if (ExecuteSqlFile(fileLines)) {
+                Console.WriteLine($"{type} seeded in {stopwatch.ElapsedMilliseconds}ms");
+            } else {
+                Console.WriteLine($"Failed to seed {type}");
+            }
+            RestoreForeignKeyConstraints(foreignKeys);
+        }
     } else {
-        Console.WriteLine($"Failed to seed {type}");
+        // Skip seeding
+        Console.WriteLine($"Skipping seeding of {type}");
+    }
+}
+
+bool IsTableEmpty(string tableName) {
+    try {
+        var entityType = GetEntityType(tableName);
+        if (entityType != null) {
+            var method = typeof(DbContext).GetMethod(nameof(DbContext.Set), new Type[] { });
+            var genericMethod = method?.MakeGenericMethod(entityType);
+            var queryable = (IQueryable<object>) genericMethod?.Invoke(ms2Context, null);
+
+            bool isEmpty = !queryable?.Any() ?? true;
+            Console.WriteLine($"Table {tableName} is empty.");
+            return isEmpty;
```

Comment:
when rerunning the seed, it says "Table is empty" but isEmpty is false and then asks to replace.

Either fix the message or remove it since 

1. on first seed it'll always be empty, so w/e
2. on second seed, the message of replacing data will request for user confirmation anyway

Reply:
I will move the message to the correct spot